bindsnet/network/topology.py

```diff
@@ -641,10 +652,15 @@ def __init__(self, source: Nodes, target: Nodes, nu: Optional[Union[float, Seque
 
         if self.w is None and self.sparsity is not None:
             i = torch.bernoulli(1 - self.sparsity * torch.ones(*source.shape, *target.shape))
-            v = self.wmin + (self.wmax - self.wmin) * torch.rand(*source.shape, *target.shape)[i.byte()]
+            if self.wmin == -np.inf or self.wmax == np.inf:
+                v = torch.clamp(torch.rand(*source.shape, *target.shape)[i.byte()], self.wmin, self.wmax)
+            else:
+                v = self.wmin + torch.rand(*source.shape, *target.shape)[i.byte()] * (self.wmax - self.wmin)
             self.w = torch.sparse.FloatTensor(i.nonzero().t(), v)
-        elif self.w is not None:
+        elif self.w is not None and self.sparsity is None:
             assert self.w.is_sparse, 'Weight matrix is not sparse (see torch.sparse module)'
+            if self.wmin != -np.inf or self.wmax != np.inf:
```

Comment:
I think one should check for weight bounds here as well

Reply:
Yup.